tests/CMakeLists.txt

```diff
@@ -40,58 +40,53 @@ set(TESTS_SRC_UTILS
         SparseGridTest
         StringTest
         UnionFindTest
-)
+        )
 
 set(TESTS_HELPERS_SRC ReadTestPolygons.cpp)
 
-if(ENABLE_ARCUS)
-    list(APPEND TESTS_SRC
-            arcus/ArcusCommunicationTest.cpp
-            arcus/ArcusCommunicationPrivateTest.cpp)
+set(TESTS_SRC_ARCUS)
+if (ENABLE_ARCUS)
+    list(APPEND TESTS_SRC_ARCUS
+            arcus/ArcusCommunicationTest
+            arcus/ArcusCommunicationPrivateTest)
     list(APPEND TESTS_HELPERS_SRC arcus/MockSocket.cpp)
-endif()
+endif ()
+
 add_library(test_helpers ${TESTS_HELPERS_SRC})
 target_link_libraries(test_helpers PRIVATE _CuraEngine GTest::gtest GTest::gmock clipper::clipper)
-if(ENABLE_ARCUS)
-    target_link_libraries(test_helpers PRIVATE arcus::libarcus protobuf::libprotobuf)
-endif()
+if (ENABLE_ARCUS)
+    target_link_libraries(test_helpers PUBLIC arcus::libarcus protobuf::libprotobuf)
+endif ()
 
-foreach(test ${TESTS_SRC_BASE})
+foreach (test ${TESTS_SRC_BASE})
     add_executable(${test} main.cpp ${test}.cpp)
     add_test(NAME ${test} COMMAND "${test}" WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
     target_link_libraries(${test} PRIVATE _CuraEngine test_helpers GTest::gtest GTest::gmock clipper::clipper)
-    if(ENABLE_ARCUS)
-        target_link_libraries(${test} PRIVATE arcus::libarcus protobuf::libprotobuf)
-    endif()
-    gtest_discover_tests(${test} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-endforeach()
+endforeach ()
+
+foreach (test ${TESTS_SRC_ARCUS})
+    string(LENGTH ${test} test_len)
+    math(EXPR test_len "${test_len} - 6")
+    string(SUBSTRING ${test} 6 ${test_len} test_name)
```

Comment:
Could you document what this does and why it is necessary?

It seems to limit the test executable length to 6 characters?

Reply:
remove the `arcus/` from the source file list, I will update this to make it the same as the other tests that are located in a sub folder. I just now noticed that there is a diffrence